src/wallet/export/caravan.rs

```diff
@@ -0,0 +1,461 @@
+                    key_source = Some((fp, kp))
+                };
+                let derivation_path =
+                    DerivationPath::master().child(ChildNumber::Normal { index: 0 });
+                k.xpub
+                    .into_descriptor_key(key_source, derivation_path)
+                    .map_err(|e| DescriptorError::Key(e))
+            })
+            .collect();
+        result
+    }
+
+    fn parse_sorted_multi<Pk: MiniscriptKey, Ctx: ScriptContext>(
+        sorted_multi: &SortedMultiVec<Pk, Ctx>,
+    ) -> (Quorum, Vec<Pk>) {
+        let quorum = Quorum {
+            required_signers: sorted_multi.k,
+            total_signers: sorted_multi.pks.len(),
+        };
+        let extended_public_keys = sorted_multi.pks.clone();
+        (quorum, extended_public_keys)
+    }
+
+    /// Export BDK wallet configuration as a Caravan configuration
+    pub fn export(
+        name: String,
+        client_type: String,
+        network: Network,
+        descriptor: &Descriptor<DescriptorPublicKey>,
+    ) -> Result<Self, Error> {
+        let (address_type, quorum, descriptor_public_keys) = match descriptor {
+            Descriptor::Sh(sh) => match sh.as_inner() {
+                ShInner::SortedMulti(smv) => {
+                    let (quorum, extended_public_keys) = CaravanExport::parse_sorted_multi(smv);
+                    Ok((CaravanAddressType::P2sh, quorum, extended_public_keys))
+                }
+                ShInner::Wsh(wsh) => match wsh.as_inner() {
+                    WshInner::SortedMulti(smv) => {
+                        let (quorum, extended_public_keys) = CaravanExport::parse_sorted_multi(smv);
+                        Ok((CaravanAddressType::P2shP2wsh, quorum, extended_public_keys))
+                    }
+                    _ => Err(Error::Generic(
+                        "Unsupported sh(wsh()) inner descriptor.".to_string(),
+                    )),
+                },
+                _ => Err(Error::Generic(
+                    "Unsupported sh() inner descriptor.".to_string(),
+                )),
+            },
+            Descriptor::Wsh(sh) => match sh.as_inner() {
+                WshInner::SortedMulti(smv) => {
+                    let (quorum, extended_public_keys) = CaravanExport::parse_sorted_multi(smv);
+                    Ok((CaravanAddressType::P2wsh, quorum, extended_public_keys))
+                }
+                _ => Err(Error::Generic(
+                    "Unsupported wsh() inner descriptor.".to_string(),
+                )),
+            },
+            _ => Err(Error::Generic(
+                "Unsupported top level descriptor.".to_string(),
+            )),
+        }?;
+
+        let network = match network {
+            Network::Bitcoin => CaravanNetwork::Mainnet,
+            _ => CaravanNetwork::Testnet,
+        };
+        let client = CaravanClient { value: client_type };
+        let extended_public_keys: Vec<CaravanExtendedPublicKey> = descriptor_public_keys
```

Comment:
nit: you don't have to specifiy the type here, the compiler can probably infer it.